src/cmd/cli/command/commands.go

```diff
@@ -545,14 +552,17 @@ var generateCmd = &cobra.Command{
 			return err
 		}
 
-		if client.CheckLoginAndToS(cmd.Context()) != nil {
+		if client.CheckLoginAndToS(cmd.Context()) != nil && !cliClient.TermsAccepted() {
 			// The user is either not logged in or has not agreed to the terms of service; ask for agreement to the terms now
 			if err := cli.InteractiveAgreeToS(cmd.Context(), client); err != nil {
-				// This might fail because the user did not log in. This is fine: we won't persist the terms agreement, but can proceed with the generation
+				// This might fail because the user did not log in. This is fine: server won't save the terms agreement, but can proceed with the generation
 				if connect.CodeOf(err) != connect.CodeUnauthenticated {
 					return err
 				}
-				// TODO: persist the terms agreement in the state file
+				// Persist the terms agreement in the state file so that we don't ask again
+				if err := cliClient.AcceptTerms(); err != nil {
+					term.Debug("unable to persist terms agreement:", err) // not fatal
+				}
```

Comment:
I think I would hide this call to `cliClient.AcceptTerms` within `InteractiveAgreeToS`, to treat the local client state more like a write-through cache.